packages/dds/tree/src/test/events/events.spec.ts

```diff
@@ -124,38 +147,43 @@ describe("EventEmitter", () => {
 		assert.strictEqual(count, 0);
 	});
 
-	// Note: This behavior is not contractually required (see docs for `Listenable.on()`),
-	// but is tested here to check for changes or regressions.
-	it("correctly handles multiple registrations of the same listener", () => {
+	it("errors on multiple registrations of the same listener", () => {
 		const emitter = createEmitter<TestEvents>();
-		let count: number;
+		let count = 0;
 		const listener = () => (count += 1);
-		const off1 = emitter.on("open", listener);
-		const off2 = emitter.on("open", listener);
-
-		count = 0;
-		emitter.emit("open");
-		assert.strictEqual(count, 2); // Listener should be fired twice
-
-		count = 0;
-		off1();
+		emitter.on("open", listener);
+		assert.throws(
+			() => emitter.on("open", listener),
+			(e: Error) => validateAssertionError(e, /register.*twice.*open/),
+		);
+		// If error is caught, the listener should still fire once for the first registration
 		emitter.emit("open");
 		assert.strictEqual(count, 1);
+	});
 
-		count = 0;
-		off2();
-		emitter.emit("open");
-		assert.strictEqual(count, 0);
+	it("allows registrations of event names that are symbols", () => {
+		const eventSymbol = Symbol("TestEvent");
+		const emitter = createEmitter<{ [eventSymbol]: () => void }>();
+		let count = 0;
+		const listener = () => (count += 1);
+		emitter.on(eventSymbol, listener);
+		emitter.emit(eventSymbol);
+		assert.equal(count, 1);
+		assert.throws(
+			() => emitter.on(eventSymbol, listener),
+			(e: Error) => validateAssertionError(e, /register.*twice.*TestEvent/),
+		);
 	});
 
 	it("allows repeat deregistrations", () => {
 		const emitter = createEmitter<TestEvents>();
 		const deregister = emitter.on("open", () => {});
-		const deregisterB = emitter.on("open", () => {});
+		const listenerB = () => {};
+		emitter.on("open", listenerB);
 		deregister();
 		deregister();
-		deregisterB();
-		deregisterB();
+		emitter.off("open", listenerB);
+		emitter.off("open", listenerB);
 	});
 
 	it("skips events adding during event", () => {
```

Comment:
Code was altered (just slightly) for when there is a deregistration during emit, is there coverage for that case like there is for adding listener?

Reply:
Good question. I didn't see any current coverage so I added a test case for it. I also added a couple more tests for some other untested features. 

I didn't intend to change the behavior, even subtly - what did I miss? Or did you mean I changed the code structure but not actually the behavior.

Reply:
I meant the latter. Behavior looks intact to me... but we can only be confident with coverage :)